torch_xla/csrc/cross_replica_reduces.cpp

```diff
@@ -11,10 +11,18 @@
 #include "torch_xla/csrc/helpers.h"
 #include "torch_xla/csrc/layout_manager.h"
 #include "torch_xla/csrc/token_handler.h"
+#include "torch_xla/csrc/xla_graph_executor.h"
 
 namespace torch_xla {
 namespace {
 
+struct AllReduceToken {
+  std::shared_ptr<torch::lazy::Value> token;
+  std::mutex lock;
+};
+
+AllReduceToken g_all_reduce_token;
```

Comment:
Shouldn't we have one token per device? Under the PJRT for v3 cases, each process will have 2 thread and 1 device per thread. In that case those two thread should not share the same token.

Reply:
Good point. Completely forgot the V3 case...